### PR TITLE
Samples module

### PR DESCRIPTION
**Pull request by ExpensiveBelly**

### CODE REVIEW

File: samples/build.gradle [hunk 1]
```diff
@@ -0,0 +1,7 @@
+apply from: '../kotlin.gradle'
+
+dependencies {
+	implementation project(':types')
+    implementation "junit:junit:4.13"
```
Comment (fsbarata) [inline samples/build.gradle]: testImplementation? if so, then it's already in kotlin.gradle

File: samples/src/main/kotlin/com/github/fsbarata/functional/samples/FormValidation.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+sealed class ValidationError {
+    object InvalidMail : ValidationError()
+    object InvalidPhoneNumber : ValidationError()
+}
+
+private fun Email.toValidatedEmail(): Validation<NonEmptyList<ValidationError>, Email> =
+    when {
+        isValidEmail() -> Validation.Success(this)
+        else -> Validation.Failure(ValidationError.InvalidMail).toValidationNel()
```
Comment (fsbarata) [inline samples/src/main/kotlin/com/github/fsbarata/functional/samples/FormValidation.kt]: Can also be Validation.Failure(nelOf(InvalidMail))

File: samples/build.gradle (hunk 1 above)
Comment (ExpensiveBelly) [reply samples/build.gradle]: Done

File: samples/src/main/kotlin/com/github/fsbarata/functional/samples/FormValidation.kt (hunk 2 above)
Comment (ExpensiveBelly) [reply samples/src/main/kotlin/com/github/fsbarata/functional/samples/FormValidation.kt]: Done

**Comment by ExpensiveBelly** [comment]
- [x] Use codestyle with tabs

**Review by fsbarata — APPROVED** [review]

**Pull request merged**